applications/callflow/src/module/cf_temporal_route.erl

```diff
@@ -179,17 +179,36 @@ get_temporal_rules([{Route, Id}|Routes], LSec, AccountDb, TZ, Now, Rules) ->
             maybe_build_rule(Routes, LSec, AccountDb, TZ, Now, Rules, Id, JObj)
     end.
 
+-spec check_end_date(non_neg_integer(), kz_term:ne_binary(), kz_time:datetime(), integer(), kzd_temporal_rules:doc()) -> 'future' | 'equal' | 'past'.
+check_end_date(LSec, TZ, Now, EndDateSeconds, RulesDoc) ->
+                                                % TODO, this is an workaround since the from_gregorian_seconds function are crashing on low values such as zero
+    case EndDateSeconds of
```

Comment:
Check the value for `'undefined'` instead of 0, and I would do it in a function clause:

```erlang
check_end_date(_LSec, _TZ, _Now, 'undefined', _RulesDoc) -> 'future';
check_end_date(...) ->
```